File: src/main/java/com/study/lec03/Lec03Main.java
package com.study.lec03;

import com.study.lec01.Person;

public class Lec03Main {

    public static void main(String[] args) {

    }

    public static void printAgeIfPerson(Object obj) {
        if (obj instanceof Person) {
            Person person = (Person) obj;
            System.out.println(person.getAge());
        }
    }

}

File: src/main/kotlin/com/study/kotlin/lec03/Lec03Main.kt
package com.study.kotlin.lec03

import com.study.lec01.Person

fun main() {
    val number1 = 3
    val number2: Long = number1?.toLong() ?: 0

    printAgeIfPerson(null)

    val person = Person("현진주", 32)
    println("이름: ${person.name}")

    val name = "현진주"
    val str = """
        ABC
        DEF
        GHI
        ${name}
    """.trimIndent()

    println(str)
    println(str[0])
    println(str[1])
}

fun printAgeIfPerson(obj: Any?) {
    val person = obj as? Person
    println(person?.age)
}